tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -1378,7 +1380,12 @@ bool EnableRowOptimization(mlir::lmhlo::FusionOp fusion) {
       absl::c_all_of(GetOutputOps(fusion), IsRowMajor);
 
   bool some_row_broadcasting = false;
```

Comment:
How does this work? Seemingly, this bool is never set to `true`?

Reply:
Sorry for nitpicking, could we also extract this into a function to make this even more clear?

```

bool some_row_broadcasting = ...()
```

Reply:
I tried and made a commit of this, but this make the code much longer.
It convert a one line of code to 16 lines of code.
You can check at that commit 527babe3c81d300f5b73186ee28fc99930fc0a71 to see the results.

So I made a 2nd commit that undo this and add comments instead.
If you still prefer the function version, just tell me and I'll remove the last commit.